go-controller/pkg/ovn/loadbalancer.go

```diff
@@ -1,108 +1,152 @@
-			"--columns=_uuid", "find", "load_balancer",
-			"external_ids:k8s-cluster-lb-udp=yes")
-	}
-	if err != nil {
-		return "", err
+		key = "k8s-cluster-lb-tcp"
+	} else {
+		key = "k8s-cluster-lb-udp"
 	}
-	if out == "" {
-		return "", fmt.Errorf("no load-balancer found in the database")
+
+	lbs := ovn.ovnNbCache.GetMap("Load_Balancer", "uuid")
+	for _, lb := range lbs {
+		externalIds := lb.(map[string]interface{})["external_ids"]
+		if externalIds == nil {
+			continue
+		}
+
+		externalID := externalIds.(map[string]interface{})[key]
+		if externalID != nil && externalID.(string) == "yes" {
+			ovn.loadbalancerClusterCache[string(protocol)] = lb.(map[string]interface{})["uuid"].(string)
+			return lb.(map[string]interface{})["uuid"].(string), nil
+		}
 	}
-	ovn.loadbalancerClusterCache[string(protocol)] = out
-	return out, nil
+
+	ovn.loadbalancerClusterCache[string(protocol)] = ""
+	return "", nil
 }
 
 func (ovn *Controller) getDefaultGatewayLoadBalancer(protocol kapi.Protocol) string {
 	if outStr, ok := ovn.loadbalancerGWCache[string(protocol)]; ok {
 		return outStr
 	}
 
-	var gw string
-	gw, _, _ = util.RunOVNNbctlUnix("--data=bare",
-		"--no-heading", "--columns=name", "find", "logical_router",
-		"options:lb_force_snat_ip=100.64.1.2")
-	if len(gw) == 0 {
+	var defaultGateway string
+
+	gateways := ovn.ovnNbCache.GetMap("Logical_Router", "uuid")
+	for _, gateway := range gateways {
+		options := gateway.(map[string]interface{})["options"]
+		if options == nil {
+			continue
+		}
+		option := options.(map[string]interface{})["lb_force_snat_ip"]
+		if option != nil && option.(string) == "100.64.1.2" {
+			defaultGateway = gateway.(map[string]interface{})["name"].(string)
+			break
+		}
+	}
+
+	if defaultGateway == "" {
 		logrus.Errorf("Error locating default gateway")
 		return ""
 	}
 
 	externalIDKey := string(protocol) + "_lb_gateway_router"
-	lb, _, _ := util.RunOVNNbctlUnix("--data=bare",
-		"--no-heading", "--columns=_uuid", "find", "load_balancer",
-		"external_ids:"+externalIDKey+"="+gw)
-	if len(lb) != 0 {
-		ovn.loadbalancerGWCache[string(protocol)] = lb
+	lbs := ovn.ovnNbCache.GetMap("Load_Balancer", "uuid")
+	for _, lb := range lbs {
+		externalIds := lb.(map[string]interface{})["external_ids"]
+		if externalIds == nil {
+			continue
+		}
+		externalID := externalIds.(map[string]interface{})[externalIDKey]
+		if externalID != nil && externalID.(string) == defaultGateway {
+			ovn.loadbalancerGWCache[string(protocol)] = lb.(map[string]interface{})["uuid"].(string)
+			return lb.(map[string]interface{})["uuid"].(string)
+		}
 	}
-	return lb
+
+	return ""
 }
 
 func (ovn *Controller) getLoadBalancerVIPS(
 	loadBalancer string) (map[string]interface{}, error) {
-	outStr, _, err := util.RunOVNNbctlHA("--data=bare", "--no-heading",
-		"get", "load_balancer", loadBalancer, "vips")
-	if err != nil {
-		return nil, err
+	lb := ovn.ovnNbCache.GetMap("Load_Balancer", "uuid", loadBalancer)
```

Comment:
I would suggest to rename loadBalancer to loadBalancerId to explicitly show that it is an Id and not a whole object